providers/cncf/kubernetes/src/airflow/providers/cncf/kubernetes/utils/pod_manager.py

```diff
@@ -99,6 +101,92 @@ def check_exception_is_kubernetes_api_unauthorized(exc: BaseException):
     return isinstance(exc, ApiException) and exc.status and str(exc.status) == "401"
 
 
+async def generic_watch_pod_events(
```

Comment:
Can you add some pydocs "why" this is called "generic" and not just `watch_pod_events()`? If there is no reason then I'd favor the name w/o generic.